src/MultiFormats/NetworkProtocol.cs

```diff
@@ -0,0 +1,557 @@
+            stream.WriteSomeBytes(bytes);
+        }
+    }
+
+    internal abstract class ValuelessNetworkProtocol : NetworkProtocol
+    {
+        public override void ReadValue(CodedInputStream stream)
+        {
+            // No value to read 
+        }
+
+        public override void ReadValue(TextReader stream)
+        {
+            // No value to read 
+        }
+
+        public override void WriteValue(CodedOutputStream stream)
+        {
+            // No value to write
+        }
+    }
+
+    internal class QuicNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "quic";
+        public override uint Code => 460;
+    }
+
+    internal class HttpNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "http";
+        public override uint Code => 480;
+    }
+
+    internal class HttpsNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "https";
+        public override uint Code => 443;
+    }
+
+    internal class WsNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "ws";
+        public override uint Code => 477;
+    }
+
+    internal class WssNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "wss";
+        public override uint Code => 478;
+    }
+
+    internal class Libp2PWebrtcStarNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "libp2p-webrtc-star";
+        public override uint Code => 275;
+    }
+
+    internal class Libp2PWebrtcDirectNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "libp2p-webrtc-direct";
+        public override uint Code => 276;
+    }
+
+    internal class UdtNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "udt";
+        public override uint Code => 301;
+    }
+
+    internal class UtpNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "utp";
+        public override uint Code => 302;
+    }
+
+    internal class P2PCircuitNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "p2p-circuit";
+        public override uint Code => 290;
+    }
+
+    internal abstract class DomainNameNetworkProtocol : NetworkProtocol
+    {
+        public string DomainName { get; set; }
+
+        public override void ReadValue(TextReader stream)
+        {
+            base.ReadValue(stream);
+            DomainName = Value;
+        }
+
+        public override void ReadValue(CodedInputStream stream)
+        {
+            Value = stream.ReadString();
+            DomainName = Value;
+        }
+
+        public override void WriteValue(TextWriter stream)
+        {
+            stream.Write('/');
+            stream.Write(DomainName.ToString());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [There's no need to call 'ToString()' on a string.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)